components/remote_settings/src/client.rs

```diff
@@ -48,6 +56,19 @@ impl<C: ApiClient> RemoteSettingsClient<C> {
         &self.collection_name
     }
 
+    /// Filters records based on the presence and evaluation of `filter_expression`.
+    fn filter_records(&self, records: Vec<RemoteSettingsRecord>) -> Vec<RemoteSettingsRecord> {
+        records
+            .into_iter()
+            .filter(|record| {
+                match &record.filter_expression {
```

Comment:
When and where do you set this attribute? 

Why not just have `match record.fields.get("filter_expession")` here? I don't get it...